internal/pkg/pluginengine/change.go

```diff
@@ -169,3 +182,56 @@ func handleResult(smgr statemanager.Manager, change *Change) error {
 	log.Successf("Plugin %s(%s) %s done.", change.Tool.Name, change.Tool.Plugin.Kind, change.ActionName)
 	return nil
 }
+
+// fillRefValueWithOutputs fill inputs from state
+func fillRefValueWithOutputs(smgr statemanager.Manager, options map[string]interface{}) error {
```

Comment:
Too many `if` in this function, it's hard to read.

Reply:
logic is a bit complicated, it has been explained